CMakeLists.txt

```diff
@@ -96,6 +96,17 @@ if (WITH_MPFR)
     set(HAVE_SYMENGINE_MPFR yes)
 endif()
 
+# Piranha
+set(WITH_PIRANHA no
+    CACHE BOOL "Build with Piranha")
+
+if (WITH_PIRANHA)
+    find_package(Piranha REQUIRED)
+    include_directories(${PIRANHA_INCLUDE_DIRS})
+    set(LIBS ${LIBS} pthread)
```

Comment:
@certik, yes, that's what's happening now for `ARB` and `MPFR`
```CMake
if (WITH_ARB)
    find_package(FLINT REQUIRED)
    include_directories(${FLINT_INCLUDE_DIRS})
    set(LIBS ${LIBS} ${FLINT_LIBRARIES})

    # Workaround for https://github.com/fredrik-johansson/arb/issues/24
    include_directories(${FLINT_INCLUDE_DIRS}/flint)

    set(WITH_MPFR yes)

    find_package(ARB REQUIRED)
    include_directories(${ARB_INCLUDE_DIRS})
    set(LIBS ${LIBS} ${ARB_LIBRARIES})
    set(HAVE_SYMENGINE_ARB yes)
endif()
```
We could do the same for `FLINT` as well.